### PR TITLE
MOBILE-2304: Create panel component in Mobile Kit

### PR DESCRIPTION
**Pull request by jordanross-wf**
Description
---
For the native viewer components, we need an expandable panel component that you can drag up with it snapping to specific heights.

What Was Changed
---
Added components:
- WPanelVC, WPagingPanelVC
  - WPagingPanelVC is a subclass of WPanelVC that has built in support for adding pages managed by a UIPageViewController
  - Constrained by size classes, so that when the view the panel exists in has a width larger than the cap (which can be changed from the default value), it will switch to a side panel from the vertical extendable panel
- WFAButton
  - A simple floating action button that you can set properties on including colors and icons, and use constraints

Testing
---
* Ensure unit tests pass and have proper code coverage
* Play with panel in example app, especially on Plus iPhone and iPads that support multitasking and play with orientation/screen changes
* Ensure floating action button feels natural
* Ensure PanelExample.swift is a broad example showcasing all features of WPanel
* Ensure everything in UX Spec is achievable with basic implementation

---

Please Review: @Workiva/mobile

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Raven
Number of Findings: 0

File: Source/WPanel.swift [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+                        closestSnapRatio = ratio
+                    }
+                }
+
+                // In the case of no ratios existing, force it to snap to 0.0 (hidden)
+                closestSnapRatio = closestSnapRatio ?? 0.0
+
+                // Check velocity of pan if user is flinging panel up or down
+                let yVelocity = -recognizer.velocityInView(view).y
+                if (yVelocity > 150 && currentSnapRatio > closestSnapRatio) {
```
Comment (brianblanchard-wf) [inline Source/WPanel.swift]: Can we define this velocity as a property so consumers can override it?

File: Source/WPanel.swift [hunk 2]
```diff
@@ -0,0 +1,789 @@
...
+                    if let nextRatio = getNextSnapRatio(closestSnapRatio!) {
+                        closestSnapRatio = nextRatio
+                    }
+                } else if (yVelocity < -150 && currentSnapRatio < closestSnapRatio) {
```
Comment (brianblanchard-wf) [inline Source/WPanel.swift]: Same here

**Comment by brianblanchard-wf** [comment]
One thing found in testing is the paging indicator does not update to the middle spot. For example, if there are 3 pages, when on the 2nd page, the page indicator does not show the 2nd one as active. When the third page is active, then the 3rd one is active. This only happens when swiping the panel open on iPads.

**Comment by jordanross-wf** [comment]
As far as the paging indicator bug goes, this seems to occur when you swipe hard from page 1 to page 3, or vice versa. After research into the issue, a fix is not trivial for this. This happens due to the fact that if you swipe hard past the first or last page, it does not call the delegate method didFinishAnimating, which is where we update the page index. A hacky fix for this would require tapping into the scroll view and setting a delegate and always manually checking when the scrollview ends what page it has ended up on. If we find an easy fix in the future or Apple changes didFinishAnimating to be called, we can include a fix in a new ticket

**Comment by brianblanchard-wf** [comment]
+10, great work!

**Comment by teresarevious-wf** [comment]
RM +1

 verified dependencies

**Pull request merged**